dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/FailoverClusterInvoker.java

```diff
@@ -93,6 +93,10 @@ public Result doInvoke(Invocation invocation, final List<Invoker<T>> invokers, L
                 if (e.isBiz()) { // biz exception.
                     throw e;
                 }
+                if (e.getCause() != null && InterruptedException.class.getName().equals(e.getCause().toString())) {
+                    // don`t catch interrupt exception
+                    throw new RuntimeException(e);
+                }

```

Comment:
You can choose other fault tolerance strategies, such as cluster="failfast"

Reply:
it is not confit, we need failover strategy at most time, but sometimes due to business needs, I need to interrupt thread pool tasks(which contains dubbo invoke)

Reply:
Interrupting a thread is not a common scenario, and I don't approve of modifying non-generic logic.
Based on your scenario, I think you may need a custom Cluster implementation.

Reply:
my code is like this

``` java
Future<?> future =RUNNER_EXECUTOR.submit(() -> {
    pipelineService.startTask(taskId);
});
```
``` java
//because some logic error, i need find the future and  cancel task
future.cancel();
```

i think it is a normal scenario......

Reply:
I suggest that you can first create an issue to discuss this requirement,  so that more people can see and participate in the discussion.

Reply:
maybe later, thanks for your patient reply.